### PR TITLE
Backend reformatting (AlignmentRecordLayout)

### PR DESCRIPTION
**Pull request by matthew-zhao**
Removed references to track or TrackLayout in AlignmentRecordLayout and MismatchLayout. AlignmentRecordLayout now returns a Map[String, Array[ReadJson]] of AlignmentRecord information, which maps each sample ID to an array of ReadJsons (which include selective information about the CalculatedAlignmentRecord, including mismatches) and filters by sampleIDs passed into the AlignmentRecordLayout object as well.

AlignmentRecordLayoutSuite tests and VizReads have also been changed to support the new format.

### CODE REVIEW

**Comment by akmorrow13** [comment]
Jenkins, test this please.

File: mango-core/src/test/scala/org/bdgenomics/mango/layout/AlignmentRecordLayoutSuite.scala [hunk 1]
```diff
@@ -24,7 +24,7 @@ import org.bdgenomics.formats.avro.{ AlignmentRecord, Contig }
 
 class AlignmentRecordLayoutSuite extends ADAMFunSuite {
 
-  sparkTest("test correct matePairs") {
```
Comment (erictu) [inline mango-core/src/test/scala/org/bdgenomics/mango/layout/AlignmentRecordLayoutSuite.scala]: should just delete the test if we aren't dealing with matePairs anymore

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala [hunk 2]
```diff
@@ -250,6 +246,7 @@ class VizServlet extends ScalatraServlet {
               val filteredData: RDD[(ReferenceRegion, CalculatedAlignmentRecord)] =
                 AlignmentRecordFilter.filterByRecordQuality(dataOption.get.toRDD(), readQuality)
               val binSize = VizUtils.getBinSize(region, VizReads.screenSize)
+              val unfilteredAlignmentData: Map[String, Array[ReadJson]] = AlignmentRecordLayout(dataOption.get.toRDD(), sampleIds)
```
Comment (erictu) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: Is this used anywhere?

File: mango-core/src/test/scala/org/bdgenomics/mango/layout/AlignmentRecordLayoutSuite.scala (hunk 1 above)
Comment (matthew-zhao) [inline mango-core/src/test/scala/org/bdgenomics/mango/layout/AlignmentRecordLayoutSuite.scala]: Alright, Alyssa told me to comment the old tests out, though (I think)

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala (hunk 2 above)
Comment (matthew-zhao) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: It's not used anywhere currently, but I was thinking when we were ready to send to the front end, we'd use it. Currently, filteredData is what we actually use, and that is passed into MergedAlignmentRecordLayout object, while I use just AlignmentRecordLayout object.

**Comment by matthew-zhao** [comment]
Alright, sorry for the delay, but I've addressed the comments you've brought up, let me know if there are any other issues I need to take care of

**Comment by erictu** [comment]
Jenkins, test this please.

**Comment by erictu** [comment]
Thanks @matthew-zhao !!!

**Pull request merged**